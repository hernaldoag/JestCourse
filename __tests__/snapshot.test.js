import { getCharacter } from '../snapshot';
import rick from '../rick.json';


describe('Es hora de las instantaneas', () => {
    test('Snapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    });

    test('siempre fallara la instantanea', () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20),   
        }
        expect(user).toMatchSnapshot();
    });

    test('Permitimos o tenemos excepcion', () =>{
        const user = {
            id: Math.floor(Math.random() * 20),
            name: "Herny Alvarez"
        }
        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        });
    });
});



