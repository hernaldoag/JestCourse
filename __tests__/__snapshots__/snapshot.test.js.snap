// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Permitimos o tenemos excepcion 1`] = `
Object {
  "id": Any<Number>,
  "name": "Hernaldo Alvarez",
}
`;

exports[`Es hora de las instantaneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instantaneas siempre fallara la instantanea 1`] = `
Object {
  "createAt": 2021-06-01T18:37:12.529Z,
  "id": 15,
}
`;
